File: apps/mobile/src/design-system/theme/typography.ts
import { TextStyle } from 'react-native';

// fontFamily values match the bundled .ttf file/PostScript names, linked via
// `npx react-native-asset` (react-native.config.js -> ./assets/fonts).
export const fonts = {
  displayRegular: 'CormorantGaramond-Regular',
  displayMedium: 'CormorantGaramond-Medium',
  displaySemiBold: 'CormorantGaramond-SemiBold',
  displayItalic: 'CormorantGaramond-Italic',
  displayMediumItalic: 'CormorantGaramond-MediumItalic',
  displaySemiBoldItalic: 'CormorantGaramond-SemiBoldItalic',
  bodyLight: 'DMSans-Light',
  bodyRegular: 'DMSans-Regular',
  bodyMedium: 'DMSans-Medium',
  bodySemiBold: 'DMSans-SemiBold',
  monoRegular: 'IBMPlexMono-Regular',
  monoMedium: 'IBMPlexMono-Medium',
} as const;

type SerifWeight = 'regular' | 'medium' | 'semiBold';

/** Cormorant Garamond — display/headline serif. */
export function display(
  size: number,
  opts: { weight?: SerifWeight; italic?: boolean } = {},
): TextStyle {
  const { weight = 'semiBold', italic = false } = opts;
  const map: Record<SerifWeight, string> = {
    regular: italic ? fonts.displayItalic : fonts.displayRegular,
    medium: italic ? fonts.displayMediumItalic : fonts.displayMedium,
    semiBold: italic ? fonts.displaySemiBoldItalic : fonts.displaySemiBold,
  };
  return { fontFamily: map[weight], fontSize: size };
}

type SansWeight = 'light' | 'regular' | 'medium' | 'semiBold';

/** DM Sans — body/UI sans. */
export function body(size: number, weight: SansWeight = 'regular'): TextStyle {
  const map: Record<SansWeight, string> = {
    light: fonts.bodyLight,
    regular: fonts.bodyRegular,
    medium: fonts.bodyMedium,
    semiBold: fonts.bodySemiBold,
  };
  return { fontFamily: map[weight], fontSize: size };
}

type MonoWeight = 'regular' | 'medium';

/** IBM Plex Mono — uppercase micro-labels. */
export function mono(size: number, weight: MonoWeight = 'regular'): TextStyle {
  const map: Record<MonoWeight, string> = {
    regular: fonts.monoRegular,
    medium: fonts.monoMedium,
  };
  return { fontFamily: map[weight], fontSize: size };
}

File: apps/ios/ATTREQ/DesignSystem/Theme/Typography.swift
import SwiftUI

/// Weights available in the bundled Cormorant Garamond serif family.
public enum AttreqSerifWeight {
    case regular, medium, semiBold

    fileprivate func postScriptName(italic: Bool) -> String {
        switch (self, italic) {
        case (.regular, false): "CormorantGaramond-Regular"
        case (.medium, false): "CormorantGaramond-Medium"
        case (.semiBold, false): "CormorantGaramond-SemiBold"
        case (.regular, true): "CormorantGaramond-Italic"
        case (.medium, true): "CormorantGaramond-MediumItalic"
        case (.semiBold, true): "CormorantGaramond-SemiBoldItalic"
        }
    }
}

/// Weights available in the bundled DM Sans family.
public enum AttreqSansWeight {
    case light, regular, medium, semiBold

    fileprivate var postScriptName: String {
        switch self {
        case .light: "DMSans-Light"
        case .regular: "DMSans-Regular"
        case .medium: "DMSans-Medium"
        case .semiBold: "DMSans-SemiBold"
        }
    }
}

/// Weights available in the bundled IBM Plex Mono family.
public enum AttreqMonoWeight {
    case regular, medium

    fileprivate var postScriptName: String {
        switch self {
        case .regular: "IBMPlexMono-Regular"
        case .medium: "IBMPlexMono-Medium"
        }
    }
}

public extension Font {
    /// Cormorant Garamond — display/headline serif (`ATTREQ_F.display`).
    static func attreqDisplay(
        _ size: CGFloat,
        weight: AttreqSerifWeight = .semiBold,
        italic: Bool = false
    ) -> Font {
        .custom(weight.postScriptName(italic: italic), size: size)
    }

    /// DM Sans — body/UI sans (`ATTREQ_F.body`).
    static func attreqBody(
        _ size: CGFloat,
        weight: AttreqSansWeight = .regular
    ) -> Font {
        .custom(weight.postScriptName, size: size)
    }

    /// IBM Plex Mono — labels/metadata mono (`ATTREQ_F.mono`).
    static func attreqMono(
        _ size: CGFloat,
        weight: AttreqMonoWeight = .regular
    ) -> Font {
        .custom(weight.postScriptName, size: size)
    }
}
